server/src/main/java/com/broadcom/lsp/cobol/service/utils/CustomThreadPoolExecutorService.java

```diff
@@ -0,0 +1,94 @@
+package com.broadcom.lsp.cobol.service.utils;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import lombok.NonNull;
+
+import java.util.Map;
+import java.util.concurrent.*;
+
+import static java.util.Collections.singletonMap;
+
+/**
+ * {@inheritDoc}
+ *
+ * <p>Implements API for our CustomThreadPoolExecutor created using ThreadPoolExecutor and
+ * ScheduledThreadPoolExecutor.
+ *
+ * <p>The main configuration parameters are: corePoolSize, maximumPoolSize, and keepAliveTime. The
+ * pool consists of a fixed number of core threads that are kept inside all the time, and some
+ * excessive threads that may be spawned and then terminated when they are not needed anymore.
+ */
+@Singleton
+public class CustomThreadPoolExecutorService implements CustomThreadPoolExecutor {
+
+  private static final String CUSTOM_THREAD_POOL = "CUSTOM_THREAD_POOL";
+  private static final String CUSTOM_SCHEDULED_THREAD_POOL = "CUSTOM_SCHEDULED_THREAD_POOL";
+
+  private int corePoolSize;
+  private int maximumPoolSize;
+  private int keepAliveTime;
+  private int corePoolSizeForScheduledThreadPool;
+
+  private Map<String, ExecutorService> executorServiceMap = Map.of();
+  private Map<String, ScheduledExecutorService> scheduledExecutorServiceMap = Map.of();
```

Comment:
Why don't store the executors without maps?

Reply:
> Why don't store the executors without maps?
I will change it, but honestly initial solution (static class) was quite sufficient for our case. These injected services are exceeding its role.